fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1710,6 +1713,13 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
 
         # Switch to FP32 shard after backward.
         self._use_fp32_param_shard([param])
+        if self.mixed_precision and self.fp32_reduce_scatter:
```

Comment:
Sorry missed this comment before. Wonder if we can have a separate commit for main_grad related changes from the changes for wps decrease fix (P841842878).

Reply:
Split into 2 PRs (#1139 and #1140)